src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -152,6 +161,54 @@ public List<MedicalInventory> getMedicalInventoryByParams(LocalDateTime dateFrom
 	 * @throws OHServiceException
 	 */
 	public Page<MedicalInventory> getMedicalInventoryByParamsPageable(LocalDateTime dateFrom, LocalDateTime dateTo, String status, String type, int page, int size) throws OHServiceException {
+		dateFrom = LocalDateTime.of(dateFrom.toLocalDate(), LocalTime.MIN);
+		dateTo = LocalDateTime.of(dateTo.toLocalDate(), LocalTime.MAX);
 		return ioOperations.getMedicalInventoryByParamsPageable(dateFrom, dateTo, status, type, page, size);
 	}
+	
+	/**
+	 * Fetch {@link MedicalInventory} with param.
+	 * 
+	 * @param inventoryId - the {@link MedicalInventory} id.
+	 * @return {@link MedicalInventory}. It could be {@code empty}.
+	 * @throws OHServiceException
+	 */
+	public MedicalInventory getInventoryById(Integer inventoryId) throws OHServiceException {
+		return ioOperations.getInventoryById(inventoryId);
+	}
+	
+	/**
+	 * Fetch {@link MedicalInventory} with param.
+	 * 
+	 * @param reference - the {@link MedicalInventory} reference.
+	 * @return {@link MedicalInventory}. It could be {@code empty}.
+	 * @throws OHServiceException
+	 */
+	public MedicalInventory getInventoryByReference(String reference) throws OHServiceException {
+		return ioOperations.getInventoryByReference(reference);
+	}
+	
+	/**
+	 * Verify if the object is valid for CRUD and return a list of errors, if any.
+	 *
+	 * @param medInventory
+	 * @throws OHDataValidationException
```

Comment:
The method signature has `OHServiceException` and the code throws `OHDataValidationException1